src/OrchardCore.Modules/OrchardCore.BackgroundTasks/Models/BackgroundTaskDocument.cs

```diff
@@ -1,8 +1,11 @@
+using System.Text.Json.Serialization;
 using OrchardCore.Data.Documents;
+using OrchardCore.Json.Serialization;
 
 namespace OrchardCore.BackgroundTasks.Models;
 
 public class BackgroundTaskDocument : Document
 {
+    [JsonConverter(typeof(CaseInsensitiveDictionaryConverter<BackgroundTaskSettings>))]
```

Comment:
You see how these properties already had a dictionary with the correct casing? I believed that's a setting on the JSON options to prefer using the existing instance instead of creating a new one (which is what the attribute does).

Reply:
https://github.com/dotnet/runtime/blob/main/src/tasks/MonoTargetsTasks/JsonToItemsTaskFactory/JsonToItemsTaskFactory.cs#L344

Reply:
@MikeAlhayek Nice find. Unfortunately it's only for `Dictionary<string, string>` instead of `Dictionary<string, TValue>`.

Reply:
@sarahelsaig do you think https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.preferredobjectcreationhandling?view=net-8.0#system-text-json-jsonserializeroptions-preferredobjectcreationhandling would be the setting to set to get parity with the previous behavior? Which is a perfectly fine default IMO.

Reply:
That's [already set](https://github.com/OrchardCMS/OrchardCore/blob/17f72c54bcdaee78a3819ba0a252cfdb745f9dca/src/OrchardCore/OrchardCore.Abstractions/Json/JOptions.cs#L24) in `JOptions.Base`. So looks like it's not enough by itself.